graphene/src/vec2.rs

```diff
@@ -2,35 +2,36 @@
 
 use crate::Vec2;
 use glib::translate::*;
+use std::fmt;
 
 impl Vec2 {
     #[doc(alias = "graphene_vec2_init")]
-    pub fn new(x: f32, y: f32) -> Vec2 {
+    pub fn new(x: f32, y: f32) -> Self {
         assert_initialized_main_thread!();
         unsafe {
-            let mut vec = Vec2::uninitialized();
+            let mut vec = Self::uninitialized();
             ffi::graphene_vec2_init(vec.to_glib_none_mut().0, x, y);
             vec
         }
     }
 
     #[doc(alias = "graphene_vec2_init_from_vec2")]
     #[doc(alias = "new_from_vec2")]
-    pub fn from_vec2(src: &Vec2) -> Vec2 {
+    pub fn from_vec2(src: &Vec2) -> Self {
         assert_initialized_main_thread!();
         unsafe {
-            let mut vec = Vec2::uninitialized();
+            let mut vec = Self::uninitialized();
             ffi::graphene_vec2_init_from_vec2(vec.to_glib_none_mut().0, src.to_glib_none().0);
             vec
         }
     }
 
     #[doc(alias = "graphene_vec2_init_from_float")]
     #[doc(alias = "new_from_float")]
-    pub fn from_float(src: &[f32; 2]) -> Vec2 {
+    pub fn from_float(src: &[f32; 2]) -> Self {
```

Comment:
Why is this taking a reference but e.g. in `Triangle` it takes the values directly? Should probably make that consistent. For these things directly the values is probably nicest to use.

Reply:
```suggestion
    pub fn from_float(src: [f32; 2]) -> Self {
```